packages/schema/src/config/experimental.ts

```diff
@@ -89,7 +89,12 @@ export default defineUntypedSchema({
       async $resolve (val, get) {
         // TODO: remove in v3.10
         val = val ?? await (get('experimental') as Promise<Record<string, any>>).then((e: Record<string, any>) => e?.inlineSSRStyles)
-        if (val === false || (await get('dev')) || (await get('ssr')) === false || (await get('builder')) === '@nuxt/webpack-builder') {
+        const [dev, ssr, builder] = await Promise.all([
```

Comment:
this seems less performant in the most common case (running in dev mode) as previously we short-circuited if any of those values were true.

Reply:
True, this might be a bit less performant in best case scenario, but it's difficult to make it more performant for the worst case scenario without sacrificing a tiny bit. Due to it being parallelized, the potential impact in best case justified the potential improvement in worst case

Reply:
Alright, the benchmark seems to show that the second approach is much slower. (difference 10 seconds). That was my mistake, I apologize. Here's the benchmark I used:
```ts
async function get(key) {
  return new Promise((resolve) => {
    setTimeout(() => {
      const values = {
        dev: Math.random() > 0.5,
        ssr: Math.random() > 0.5,
        builder: '@nuxt/webpack-builder'
      };
      resolve(values[key]);
    }, 100);
  });
}

async function snippet1(val) {
  if (val === false || (await get('dev')) || (await get('ssr')) === false || (await get('builder')) === '@nuxt/webpack-builder') {
    return true;
  }
  return false;
}

async function snippet2(val) {
  const [dev, ssr, builder] = await Promise.all([
    get('dev'),
    get('ssr'),
    get('builder'),
  ]);
  if (val === false || dev || ssr === false || builder === '@nuxt/webpack-builder') {
    return true;
  }
  return false;
}

async function benchmark() {
  const iterations = 100;
  let start, end;
  
  console.time("e")
  for (let i = 0; i < iterations; i++) {
    await snippet1(false);
  }
  console.timeEnd("e")
  
  console.time("F")
  for (let i = 0; i < iterations; i++) {
    await snippet2(false);
  }
  console.timeEnd("F")
}
```
![image](https://github.com/nuxt/nuxt/assets/92037085/744a817f-5ace-412d-af73-929972bfa9d9)

Reply:
Actually, when testing with val = true, I get that the new version is almost twice as fast:
![image](https://github.com/nuxt/nuxt/assets/92037085/a981aa0f-7e2e-46a9-bf05-b3f9751cc869)

Reply:
I pushed an update that combines the best of both worlds. If val is false it will immediately return false before awaiting the Promise.all, as per the results of the benchmark when val is set to false.
Otherwise it awaits promise.all and handles everything else. This actually doesn't impact the best case scenario at all, when dev is true
![image](https://github.com/nuxt/nuxt/assets/92037085/d414bba1-73ce-4387-835d-1e106bb67f2c)

And it does make sense when I think about it, considering Promise.all parallelizes it, so each promise is handled in a different core, separate from the rest. There is no dependency between each promise so there's nothing to cause slowdown. Of course, as the if conditions progress to the next, the performance difference becomes much bigger (for example, here it is when dev is hardcoded to false)
![image](https://github.com/nuxt/nuxt/assets/92037085/6f003583-167d-48ab-8ebf-611ab0ebf2ae)

Reply:
I don't think your benchmark is right. It has an arbitrary delay of 100ms for any `get` operation, which is not accurate, and doesn't account for the fact that this is only running _once_ when starting Nuxt.

In general, I would rather not waste too much time optimising something that only runs once if it adds complexity to the code.

And in the case I'm not sure it's the right 'happy-path' because the vast majority of calls to resolve the schema is running the dev server.